examples/inputs/gch_multiple_turbine_types.yaml

```diff
@@ -0,0 +1,94 @@
+
+name: GCH
+description: Three turbines using Gauss Curl Hybrid model
+floris_version: v3.0.0
+
+logging:
+  console:
+    enable: false
+    level: WARNING
+  file:
+    enable: false
+    level: WARNING
+
+solver:
+  type: turbine_grid
+  turbine_grid_points: 5
+
+farm:
+  layout_x:
+  - 0.0
+  - 630.0
+  - 1260.0
+  layout_y:
+  - 0.0
+  - 0.0
+  - 0.0
+  turbine_type:
+  - nrel_5MW
+  - nrel_5MW
+  - iea_10MW
+  turbine_definitions:
+    nrel_5MW: !include turbine_definitions/nrel_5MW.yaml
```

Comment:
I really like this! I think this makes it easier for users to see the different turbines implemented in `floris` and what is needed to implement their own turbine.

My only thought is whether something like `layout_z` or `turbine_height` should be added to `gch_multiple_turbine_types.yaml` to allow users to specify the center point of the turbine rotor. For example:

```
farm:
  layout_x:
  - 0.0
  - 630.0
  - 1260.0
  layout_y:
  - 0.0
  - 0.0
  - 0.0
  layout_z:
  - 90.0
  - 90.0
  - 119.0
  turbine_type:
  - nrel_5MW
  - nrel_5MW
  - iea_10MW
  turbine_definitions:
    nrel_5MW: !include turbine_definitions/nrel_5MW.yaml
    iea_10MW: !include turbine_definitions/iea_10MW.yaml
```

and then remove `hub_height: 90.0` and `hub_height: 119.0` from the `nrel_5MW.yaml` and `iea_10MW.yaml` files. Namely, I know working with TU Munich on the MATLAB version of `floris` that they specifically requested this. I think they were working with farms with the same turbine types but different tower lengths, and this would allow them to easily implement this without having to make multiple copies of the same turbine definition. Additionally, it would resolve issue #224.

Reply:
Thanks for the feedback Bart! I've been thinking about this more, and I think that the most probable use case by our users would be to use the hub height defined in the turbine definition, with the varying hub heights being more of a special case for optimization or trying to represent complex terrain. It is something I definitely want to support! But don't want to require the normal user to have to also enter the same hub height value for a homogenous turbine-type farm (e.g. having 100 entries of `layout_z` for a 100 turbine farm would add clutter to the input file). As such, what I am thinking is to allow the user to define `layout_z`, but not require it, and in the cases where it is defined, it will overwrite the hub height values in the turbine definition files. This has some implications on the back end where we use the hub heights in the wake calculations, and isn't immediately trivial to implement. Thus I propose addressing this in v3.1, and have made such a comment on the related issue you shared.

Reply:
That makes a lot of sense. I agree with you that it'll be a niche use case and like your proposed solution. Thanks Chris!